### ISSUE TITLE
Disallowing use of the Object() constructor without `new`

### ISSUE THREAD

**Issue by Zirro** [eslint/eslint#10243]
**What rule do you want to change?**
Between the rules [`no-array-constructor`](https://github.com/eslint/eslint/blob/master/docs/rules/no-array-constructor.md) and [`no-new-object`](https://github.com/eslint/eslint/blob/master/docs/rules/no-new-object.md), the case of an object constructor without `new` - just `Object()` - seems to have slipped through the cracks.

Since the rule `no-array-constructor` disallows both `Array()` and `new Array()`, it seems like it would be more consistent if the `no-new-object` rule could be renamed `no-object-constructor` and disallow both `Object()` and `new Object()` as well. I would understand if something less intrusive than renaming the rule might be preferable, though.

Another option might be to make the `no-array-constructor` generic so that it could apply to both arrays and objects.

**Does this change cause the rule to produce more or fewer warnings?**
More. In addition to disallowing `new Object()`, it would also disallow `Object()`.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behaviour presumably.

**Please provide some example code that this change will affect:**

```js
const list = Object();
```

**What does the rule currently do for this code?**
The mentioned rules do not currently apply to this code.

**What will the rule do after it's changed?**
It will produce an error when the `Object()` constructor is used, like it does for `new Object()` today. The intention is to encourage use of the object literal `{}` instead.

**Comment by platinumazure** [comment]
I'm not opposed to the change (although it is a breaking change).

But I do want to point out that you can currently lint for this using [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax):

```js
/* eslint no-restricted-syntax: [error, "CallExpression[callee.name='Object']"] */
Object();
```

Although it does have the small risk of false positives if someone shadows `Object` with a local variable.

**Comment by mysticatea** [comment]
We can add `no-object-constructor` like `no-array-constructor` then deprecate `no-new-object`. In that case, this is not a breaking change. As a note, `Object(x)` is legal expression to get wrapper objects.

**Comment by Mouvedia** [comment]
> The intention is to encourage use of the object literal {} instead.

That's it? Sounds like a stylistic preference to me.
What would you replace `foo === Object(foo)` with?
